torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2678,6 +2759,32 @@ std::shared_ptr<NCCLComm> ProcessGroupNCCL::initNCCLComm(
   }
 #endif // NCCL_HAS_COMM_SPLIT
 
+#ifdef NCCL_HAS_INIT_RANK_SCALABLE
+  // (nranks / nroots) == 128 was the default NCCL recommended
+  // accoring to
+  // https://github.com/pytorch/pytorch/pull/136789#discussion_r1779171615.
+  auto ranksPerRoot = getCvarInt(TORCH_NCCL_SCALABLE_INIT_RANKS_PER_ROOT, 128);
+  auto numRoots = (getSize() / ranksPerRoot) + (getSize() % ranksPerRoot > 0);
```

Comment:
nit: `(getSize() + ranksPerRoot - 1) / ranksPerRoot`

Reply:
Oh I forgot this way...